draft-gpew-priv-ppm.md

```diff
@@ -900,37 +900,70 @@ The named parameters are:
 * `batch_interval`, the request's batch interval.
 * `agg_param`, an aggregation parameter for the VDAF being executed.
 
-Depending on the VDAF scheme and how the leader is configured, the collect
-request may cause the leader to send a series of aggregate requests to the
-helpers in order to compute their share of the aggregate. Alternately, the
-leader may already have made these requests and can respond immediately. In
-either case it responds to the collector's request as follows.
+Depending on the VDAF scheme and how the leader is configured, the leader and
+helper may already have prepared all the reports falling within `batch_interval`
+and be ready to return the aggregate shares right away, but this cannot be
+guaranteed and in some VDAFs, it may not be possible to begin preparing inputs
+until the collector provides the aggregation parameter in the `CollectReq`. For
+those reasons, collect requests are handled asynchronously.
+
+Upon receipt of a `CollectReq`, the leader begins by checking that the request
+meets the requirements of the batch parameters using the procedure in
+{{batch-parameter-validation}}. If so, it immediately sends the collector a
+response with HTTP status 202 Accepted. The body consists of a `CollectResp`:
 
-It begins by checking that the request meets the requirements of the batch
-parameters using the procedure in {{batch-parameter-validation}}. If so, it
-obtains the helper's encrypted aggregate share for the batch interval by sending
-an AggregateShareReq to the helper as described in {{aggregate-share-request}}.
-(This request may too have been made in advance.)
-
-Next, the leader computes its own aggregate share by aggregating all of the
-verified output shares that fall within the batch interval. Finally, it responds
-with HTTP status 200 and a body consisting of a CollectResp message:
-
-[OPEN ISSUE: What happens if this all takes a really long time.]
+~~~
+struct {
+  Url collect_job;
+} CollectResp;
+~~~
```

Comment:
Is sending a binary-encoded response with the URL necessary/desirable? What about just adding the URL as an HTTP header? What does ACME do here?

Reply:
I would add a TODO just to get this done.